projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp

```diff
@@ -2244,6 +2347,16 @@ tool_init(rocprofiler_client_finalize_t fini_func, void* tool_data)
         start_context(counter_collection_ctx, "counter collection");
     }
 
+    if(tool::get_config().spm_counter_collection)
+    {
+        ROCPROFILER_CALL(
+            rocprofiler_configure_spm_dispatch_service(
+                get_client_ctx(), spm_dispatch_callback, nullptr, spm_data_callback, nullptr),
```

Comment:
For counter collection, we create a separate context. Any reason why we aren't doing that here?